### ISSUE TITLE
Migrate to github actions

### ISSUE THREAD

**Issue by niknetniko** [ZeusWPI/hydra#345]
Travis is becoming a PITA with their limits for open source and long build queues.

**Comment by redfast00** [comment]
I agree that we should move away from Travis, however, won't we have the same problem with Actions? Is there an open implementation of Actions so we can run them ourselves if GitHub decides to discontinue them?

**Comment by niknetniko** [comment]
> I agree that we should move away from Travis, however, won't we have the same problem with Actions? Is there an open implementation of Actions so we can run them ourselves if GitHub decides to discontinue them?

I'm not opposed, but at the moment Actions is very much sufficient for our use in this repo, while Travis has:
- Severely limited capacity on travis.org, meaning long queue times
- Travis.org will be shutdown in the near future
- On Travis.com, we'll need to apply for an allotment of OSS minutes, which must be manually granted, and some reports say they've stopped granting those.

Some info on Travis changes, from https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing:

> - For those of you who have been building on public repositories (on travis-ci.com, with no paid subscription), we will upgrade you to our trial (free) plan with a 10K credit allotment (which allows around 1000 minutes in a Linux environment).
> - You will not need to change your build definitions when you are pointed to the new plan
> - When your credit allotment runs out - we’d love for you to consider which of our plans will meet your needs.
> - We will be offering an allotment of OSS minutes that will be reviewed and allocated on a case by case basis. Should you want to apply for these credits please open a request with Travis CI support stating that you’d like to be considered for the OSS allotment. Please include:
>    - Your account name and VCS provider (like travis-ci.com/github/[your account name] )
>    - How many credits (build minutes) you’d like to request (should your run out of credits again you can repeat the process to request more or discuss a renewable amount)
> Usage will be tracked under your account information so that you can better understand how many credits/minutes are being used

So at the moment the fastest option is to move to GitHub Actions.
We could look into using our own self-hosted runners: https://docs.github.com/en/free-pro-team@latest/actions/hosting-your-own-runners/about-self-hosted-runners

**Comment by redfast00** [comment]
Just saw that this is the hydra repository, not the zeus.ugent.be repository. As project lead, if you decide GitHub Actions are better than Travis, go ahead!

**Comment by niknetniko** [comment]
> Just saw that this is the hydra repository, not the zeus.ugent.be repository. As project lead, if you decide GitHub Actions are better than Travis, go ahead!

It's not that urgent, so I'm willing to wait if there is a plan for a Zeus-hosted alternative (this will also be relevant for zeus.ugent.be in the not-so-distant future)

**Comment by redfast00** [comment]
There's not currently, but I think Gitlab runners are feasible now on our new container infrastructure? This is all still very vague though, no decisions have been made yet.

**Comment by FKD13** [comment]
We are not planing for a Zeus hosted alternative at the moment, so I would say we move this to Actions.
